libs/ledger-live-common/src/families/elrond/banner.ts

```diff
@@ -0,0 +1,106 @@
+import type { ElrondAccount, ElrondProvider } from "./types";
+import { ElrondDelegation, ElrondPreloadData } from "./types";
+import { ELROND_LEDGER_VALIDATOR_ADDRESS } from "./constants";
+
+type MappedElrondDelegation = ElrondDelegation & {
+  validator: ElrondProvider | undefined;
+};
+
+interface AccountBannerHiddenState {
+  bannerType: "hidden";
+}
+
+interface AccountBannerDelegateState {
+  bannerType: "delegate";
+}
+
+interface AccountBannerRedelegateState {
+  bannerType: "redelegate";
+  mappedDelegations: MappedElrondDelegation[];
+  selectedDelegation: ElrondDelegation;
+}
+
+export type AccountBannerState =
+  | AccountBannerDelegateState
+  | AccountBannerRedelegateState
+  | AccountBannerHiddenState;
+
+export function getAccountBannerState(
+  account: ElrondAccount,
+  elrondPreloadData: ElrondPreloadData
+): AccountBannerState {
+  const elrondResources = account.elrondResources
+    ? account.elrondResources
+    : { delegations: [] as ElrondDelegation[] };
+
+  const { validators } = elrondPreloadData;
+
+  const delegationAddresses = elrondResources.delegations.map((delegation) => {
+    return delegation.address;
+  });
+
+  if (!account.balance.isZero() && delegationAddresses.length === 0) {
+    return {
+      bannerType: "delegate",
+    };
+  }
+
+  if (delegationAddresses.length > 0 && account.balance.isZero()) {
+    return {
+      bannerType: "hidden",
+    };
+  }
```

Comment:
Can we use the same criteria than AccountHeaderManageAction is using? 
```
earnRewardEnabled && hasDelegations
```
(We can move those function to common and use them in LLM in the future.)

Reply:
Yes, happy to use those functions. So will this then look like this?

```
if (!(earnRewardsEnabled && hasDelegation)) { return { bannerType: "hidden" } }
```

Reply:
As "redelegation" has preference...I am not sure if we must move this logic to the button, to simplify it.

```
 if (hasDelegation) {
{...}
  if (worstDelegator !== ledger) {
     return redelegate
  }
  if (earnRewardsEnabled) {
    return delegate
  } else {
    return hidden
  }
} 
```

Reply:
https://github.com/LedgerHQ/ledger-live/pull/2672/commits/e2109f0e00214e2c5465f6008f16f9c71c3e0c5f